sherpa/astro/utils/smoke.py

```diff
@@ -0,0 +1,124 @@
+#
+#  Copyright (C) 2016  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+import unittest
+from tempfile import NamedTemporaryFile
+
+from sherpa.utils import requires_fits, requires_xspec
+from sherpa.astro import ui
+from numpy.testing import assert_almost_equal
+import logging
+import sys
+import os
+
+logger = logging.getLogger("sherpa")
+
+
+def run(verbosity=0, require_failure=False, fits=None, xspec=False):
+    if fits:
+        try:
+            __import__(fits, globals(), locals())
+        except ImportError:
+            _import_error("fits", fits)
+
+    if xspec:
+        try:
+            import sherpa.astro.xspec
+        except ImportError:
+            _import_error("xspec", "xspec")
+
+    test_suite = SmokeTestSuite(require_failure=require_failure)
+    runner = unittest.TextTestRunner(verbosity=verbosity)
+    result = runner.run(test_suite)
+    if result is None or result.failures or result.errors or result.unexpectedSuccesses:
+        sys.exit("Test failures were detected")
+
+
+class SmokeTest(unittest.TestCase):
+    def setUp(self):
```

Comment:
Somewhere I think we need to specify what we mean by a "smoke test of Sherpa" - i.e. what level of functionality is being tested. I don't know if it's here, the `smoke` module, or just in the documentation (but I think it would be good to have something in the code, close to where the tests are, so that future readers understand the scope).